pkg/translate/translate_value.go

```diff
@@ -325,6 +325,13 @@ apiVersion: apps/v1
 kind: Deployment
 name: istio-%s`
 
+	// need to do special handling for gateways and mixer
+	// ex. deployment name should be istio-telemetry instead of istio-mixer.telemetry
+	if newPS == "mixer.policy" || newPS == "mixer.telemetry" ||
+		newPS == "gateways.istio-ingressgateway" || newPS == "gateways.istio-egressgateway" {
+		newPS = newP[1 : len(newP)-1].String()
```

Comment:
comment above doesn't agree with what the code is doing - here were're just truncating the path, example shows translation.

Reply:
added more clear explanation